blog/_posts/2024-12-04-mirrord-on-aks.md

```diff
@@ -0,0 +1,184 @@
+
+![Screenshot of the pet store admin page](/AKS/assets/images/mirrord-on-aks/aks-pet-store-admin-page.png)
+
+Say, for example, that you have had a genius new idea regarding order spam detection: you want to try discarding orders in the queue that contain more than one type of item.
+
+Clicking on some of the orders, you can see that they’re a random collection of usually a few different types of item, like this (note that orders are created in a way such that some will have multiple entries for the same kind of item - we think these are spam too, and we only want to process orders with a single item entry):
+
+![Screenshot of the pet store orders](/AKS/assets/images/mirrord-on-aks/aks-pet-store-orders.png)
+
+We don’t want these, they’re (possibly) spam! Change the contents of `main.go` as follows: replace lines 77-78 with this code for our new anti-spam measure:
+
+```go
+// NEW: Discard orders containing more than 1 different type of item
+var new_orders []Order
+for _, order := range orders {
+	if len(order.Items) <= 1 {
+    	new_orders = append(new_orders, order)
+	}
+}
+
+// Save orders to database
+err = client.repo.InsertOrders(new_orders)
+```
+
+Re-run the debugger, refresh the `store-admin` page and check for new orders - they should all be orders with only one type of item, which you can check by clicking on the order number:
+
+![Screenshot of the pet store admin orders](/AKS/assets/images/mirrord-on-aks/aks-pet-store-admin-orders.png)
+
+Much better. This means that mirrord not only stole traffic for the `makeline-service` deployment - it also read messages from RabbitMQ and wrote new entries to MongoDB, all without messing around with configuration and credentials! It can do this because by targeting the `makeline-service` pod, it reads the environment variables and files from that pod in the cluster.
+
+
+## Summary / Further Reading
+
+In summary: if you're looking for a tool to make developing and testing Kubernetes applications easier (and faster), then give mirrord a try - it's easy and free to start using, simple to configure and powerful enough to work well with large and complex clusters. Even though the examples here are simple, mirrord has the potential to completely transform how you work on a day-to-day basis.
+
+Of course, this post only scratches the surface of what mirrord can do. You can run it in a container, you can use it for port-forwarding to the cluster (or in reverse from the cluster), you can split Kafka and SQS queues, and so on and so forth. Stay updated on the latest features through the [MetalBear blog](https://metalbear.co/blog/) or dive straight into the [mirrord docs](https://mirrord.dev/docs/overview/introduction/) to get started. Have questions? Join our friendly [Discord server](https://discord.gg/metalbear) and say hello! For additional guidance on getting started with AKS, don’t forget to check out the [AKS documentation](https://learn.microsoft.com/en-us/azure/aks/).
```

Comment:
what do you mean by "split Kafka and SQS queues"

Reply:
https://mirrord.dev/docs/using-mirrord/queue-splitting/